dataGenerator/itemData.js

```diff
@@ -0,0 +1,161 @@
+    product.price_retail = (priceSeed + 49.99).toFixed(2);
+    product.price_discount = Math.floor(Math.random() * priceSeed) + 45.97;
+  }
+  if (seed === 3) { // 20%  between $100.99 and $399.99
+    priceSeed = Math.floor(Math.random() * 99);
+    product.price_retail = (priceSeed + 99.99).toFixed(2);
+    product.price_discount = Math.floor(Math.random() * priceSeed) + 89.97;
+  }
+  product.price_discount_percent = 1 - product.price_discount / product.price_retail;
+};
+
+const getQtyPrimary = (product) => {
+  product.qty = (Math.floor(Math.random() * 101));
+};
+
+const getQtyOption = (option) => {
+  option.qty = (Math.floor(Math.random() * 5));
+};
+
+const getOptions = (product, priceSeed) => {
+  // A bunch of option types, weighted according to my whims
+  const options = [];
+  let optionTemp;
+
+  // some manual sets of common options
+  const blackandwhite = ['Color', 'White', 'Black'];
+  options.push(blackandwhite, blackandwhite); // x2
+
+  const monochromatic = ['Color', 'White', 'Black', 'Silver'];
+  options.push(monochromatic, monochromatic); // x2
+
+  const commoncolors = ['Color', 'White', 'Red', 'Blue', 'Black', 'Gray', 'Green'];
+  options.push(commoncolors, commoncolors); // x2
+
+  const lotsacolors = ['Color', 'White', 'Yellow', 'Blue', 'Red', 'Green', 'Black', 'Brown', 'Azure', 'Ivory', 'Teal', 'Silver', 'Purple', 'Navy blue', 'Pea green', 'Gray', 'Orange', 'Maroon', 'Charcoal', 'Aquamarine', 'Coral', 'Fuchsia', 'Wheat', 'Lime', 'Crimson', 'Khaki', 'Hot pink', 'Magenta', 'Plum', 'Olive', 'Cyan'];
+  options.push(lotsacolors); // x1
+
+  const tshirtsizes = ['Size', 'XS', 'S', 'M', 'L', 'XL', 'XXL', '3XL'];
+  options.push(tshirtsizes, tshirtsizes, tshirtsizes, tshirtsizes); // x4
+
+  const shoesizes = ['Size', '5', '5.5', '6', '6.5', '7', '7.5', '8', '8.5', '9', '9.5', '10', '10.5', '11', '11.5', '12', '12.5', '13', '13.5'];
+  options.push(shoesizes, shoesizes, shoesizes, shoesizes); // x4
+
+  // and 7 sets of fully random option sets
+  for (let i = 0; i < 8; i++) {
+    optionTemp = ['Options'];
+    for (let j = 0; j < Math.ceil(Math.random() * 6); j++) {
+      optionTemp.push(faker.random.words(Math.ceil(Math.random() * 3)));
+    }
+    options.push(optionTemp);
+  }
+
+  product.price_discount_percent = 0;
+  const opt = options[Math.floor(Math.random() * options.length)];
+
+  const samePrices = ((!Math.round(Math.random()))); // 50% chance all prices the same
+  product.options = opt.map((option, index) => {
+    optionTemp = { id: index, description: option };
+    getQtyOption(optionTemp);
+    getPrices(optionTemp, priceSeed);
+    if (optionTemp.price_discount_percent > product.price_discount_percent) {
+      product.default_option = index;
+      product.price_discount_percent = optionTemp.price_discount_percent;
+      product.price_retail = optionTemp.price_retail;
+      product.price_discount = optionTemp.price_discount;
+    }
+    return optionTemp;
+  });
+};
```

Comment:
I find this code hard to understand. I wonder if you're trying to do to much in one function, and if it would be better to break it up into smaller functions.

Reply:
Alternatively, you might just need to add more comments to explain what's going on.

Reply:
One other thing to note: because you have the options category listed as the first index of each array (ie, 'color', 'size', etc.), that actually gets pushed as the first option for the final product, complete with quantity, price_retail, and price_discount, etc. Is that intended?